File: TaskTracker.Model/UserTask/Request/GetUserTaskRequest.cs
using TaskTracker.Model.Attribute;
using TaskTracker.Model.Enum;

namespace TaskTracker.Model.UserTask.Request
{
    public class GetUserTaskRequest : PagedRequest
    {
        public long? CreatorId { get; set; }
        [CustomFilter(ComparisonType = ComparisonType.LesserOrEqual, PropertyName = "DueTimestamp")]
        public long? MaxDueTimestamp { get; set; }
        [CustomFilter(ComparisonType = ComparisonType.GreaterOrEqual, PropertyName = "CreationTimestamp")]
        public long? MinCreationTimestamp { get; set; }

        public long? ListId { get; set; }

        public long? StatusId { get; set; }

        [CustomFilter(ComparisonType = ComparisonType.Contains, PropertyName = "StatusId")]
        public ICollection<long>? StatusIds { get; set; }

        [CustomFilter(ComparisonType = ComparisonType.DoesNotContain, PropertyName = "StatusId")]
        public ICollection<long>? SkipStatusIds { get; set; }
    }
}

File: task-tracker-ui/src/app/model/user-task/GetUserTaskRequest.ts
import { HttpParams } from "@angular/common/http";
import PagedRequest, { mapPagedRequest } from "../PagedRequest";

export default interface GetUserTaskRequest extends PagedRequest {
    creatorId?: number,
    maxDueTimestamp?: number,
    minCreationTimestamp?: number
    listId?: number,
    skipStatusIds?: number[],
    statusIds?: number[]
}

export const mapGetUserTaskRequest = (request: GetUserTaskRequest): HttpParams => {
    let params = mapPagedRequest(request);

    if (request.creatorId) {
        params = params.append('CreatorId', request.creatorId)
    }
    if (request.maxDueTimestamp) {
        params = params.append('MaxDueTimestamp', request.maxDueTimestamp)
    }
    if (request.minCreationTimestamp) {
        params = params.append('MinCreationTimestamp', request.minCreationTimestamp)
    }
    if (request.listId) {
        params = params.append('ListId', request.listId)
    }
    if(request.skipStatusIds) {
        request.skipStatusIds.forEach(id => {
            params = params.append('SkipStatusIds', id)
        });
    }
    if(request.statusIds) {
        request.statusIds.forEach(id => {
            params = params.append('StatusIds', id)
        })
    }


    return params;
} 